play-validations/memory-footprint/src/test/java/com/google/wear/watchface/dfx/memory/AmbientMemoryFootprintCalculatorTest.java

```diff
@@ -0,0 +1,448 @@
+                        .build();
+
+        long result =
+                new AmbientMemoryFootprintCalculator(document, map, TEST_SETTINGS_NO_OLD_CLOCKS)
+                        .computeAmbientMemoryFootprint(TEST_WIDTH, TEST_HEIGHT);
+
+        // NB the second hand should not be displayed in ambient.
+        assertThat(result).isEqualTo(2 * TEST_LAYER_SIZE + 2 * singleHandSize);
+    }
+
+    @Test
+    public void computeAmbientMemoryFootprint_oldStyleAnalogClock_supported() throws Exception {
+        Document document = readDocument("OldStyleAnalogClock");
+        int singleHandSize = 300;
+        Map<String, DrawableResourceDetails> map =
+                ImmutableMap.<String, DrawableResourceDetails>builder()
+                        .put("image1", resDetails("image1", 12))
+                        .put("image2", resDetails("image1", 12))
+                        .put("hour-hand", resDetails("hour-hand", singleHandSize))
+                        .put("minute-hand", resDetails("minute-hand", singleHandSize))
+                        .build();
+
+        long result =
+                new AmbientMemoryFootprintCalculator(document, map, TEST_SETTINGS)
+                        .computeAmbientMemoryFootprint(TEST_WIDTH, TEST_HEIGHT);
+
+        // NB the second hand should not be displayed in ambient.
+        assertThat(result).isEqualTo(2 * TEST_LAYER_SIZE + 2 * singleHandSize);
+    }
+
+    @Test
+    public void computeAmbientMemoryFootprint_oldStyleAnalogClock_notSupported() throws Exception {
+        Document document = readDocument("OldStyleAnalogClock");
+        int singleHandSize = 300;
+        Map<String, DrawableResourceDetails> map =
+                ImmutableMap.<String, DrawableResourceDetails>builder()
+                        .put("image1", resDetails("image1", 12))
+                        .put("image2", resDetails("image1", 12))
+                        .put("hour-hand", resDetails("hour-hand", singleHandSize))
+                        .put("minute-hand", resDetails("minute-hand", singleHandSize))
+                        .build();
+
+        long result =
+                new AmbientMemoryFootprintCalculator(document, map, TEST_SETTINGS_NO_OLD_CLOCKS)
+                        .computeAmbientMemoryFootprint(TEST_WIDTH, TEST_HEIGHT);
+
+        // The example flattens down to a single layer.
+        assertThat(result).isEqualTo(1 * TEST_LAYER_SIZE);
+    }
+
+    @Test
+    public void computeAmbientMemoryFootprint_oldStyleDigitalClockType1_supported()
+            throws Exception {
+        Document document = readDocument("OldStyleDigitalClockType1");
+        Map<String, DrawableResourceDetails> map =
+                ImmutableMap.<String, DrawableResourceDetails>builder()
+                        .put("image1", resDetails("image1", 12))
+                        .put("image2", resDetails("image1", 12))
+                        .put("ariel.ttf", resDetails("ariel.ttf", 100))
```

Comment:
Does the test require these fonts to run?

Reply:
Nope, this is mock data that says "the ariel.ttf font has 100 bytes" for the purpose of this test.